lib/partitions_validator_utils.pm

```diff
@@ -38,7 +39,7 @@ sub validate_partition_creation {
     my @lsblk_output = split(/\n/, script_output("lsblk -n"));
     my $check;
     foreach (@lsblk_output) {
-        if ($_ =~ /(?<check>\Q$args->{mount_point}\E\s*\Z)/) {
+        if ($_ =~ /(?<check>$args->{mount_point}]?\Z)/) {
```

Comment:
I wonder if we should get rid of `\Z` instead of looking for optional square bracket. Or at least we should escape it.

Reply:
`\Z` ensure that it looks for matching at the end of the line. i think it is fine to keep it. without it if `[SWAP]` is represented twice in the line it would not get the right one.

Reply:
Then let's escape `]` and maybe even add one in from of the regexp, so it's clear that we look for something in brackets.

Reply:
This might work: `/(?<check>$args->{mount_point}[^]]*/)`

Reply:
As we don't use `/m` flag for multiline search, we could use `/\[?(?<check>$args->{mount_point})\]?$/`. Also, if we would use `lsblk -o MOUNTPOINT` we would get list of mount points only and we could just match string `SWAP` or path from test data without any risk of false positives.

Reply:
should we take into account not match 'home1' for example? Removing \Z I think it has that effect.

Reply:
With `lsblk -o MOUNTPOINT` I believe that we can set strict expectations, even with strict match of the beginning of the string using `^`.

Reply:
`/\[?(?<check>$args->{mount_point})\]?$/` is neat and with `lsblk -o MOUNTPOINT` works perfectly as expected. i adapted that. thanks. out of curiosity tho i think `/(?<check>$args->{mount_point}]?\Z)/` match the `]` literally so i got confused with you asking to escape it.

Reply:
I was not sure how it worked, maybe perl is smart enough to change behavior on the fly, but as `[]` are special symbols, it will be safe to escape them.